### PR TITLE
[D&D] Adds Bar line and Area charts to Wizard

### PR DESCRIPTION
**Pull request by ashwin-pc**
### Description
- Adds Bar line and Area charts to Wizard
- Adds resizable right nav to Wizard

TODO: Will add E2E tests before merging
 
Screen Shot 2022-09-02 at 10 00 11 PM

### Issues Resolved
- #1616 
- #1617 
- #1618
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
    - [ ] `yarn test:jest`
    - [ ] `yarn test:jest_integration`
    - [ ] `yarn test:ftr`
- [ ] New functionality has been documented.
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/wizard/public/application/_variables.scss [hunk 1]
```diff
@@ -6,4 +6,5 @@
 @import "@elastic/eui/src/global_styling/variables/form";
 
 $osdHeaderOffset: $euiHeaderHeightCompensation;
-$wizSideNavWidth: 470px;
+$wizLeftNavWidth: 462px;
+$wizRightNavWidth: 320px;
```
Comment (joshuarrrr) [inline src/plugins/wizard/public/application/_variables.scss:10]: Is this actually used?

File: src/plugins/wizard/public/application/components/right_nav.tsx [hunk 2]
```diff
@@ -32,19 +42,48 @@ export const RightNav = () => {
...
+            dispatch(
+              setActiveVisualization({
+                name: newVisType,
+                style: types.get(newVisType)?.ui.containerConfig.style.defaults,
+              })
+            );
+
+            setNewVisType(undefined);
+          }}
+          maxWidth="300px"
```
Comment (joshuarrrr) [inline src/plugins/wizard/public/application/components/right_nav.tsx:76]: nit - we should have a cleanup issue to deal with some of these hard-coded widths that are littered around in some of these .tsx files.

File: src/plugins/wizard/public/application/components/right_nav.tsx [hunk 3]
```diff
@@ -32,19 +42,48 @@ export const RightNav = () => {
...
+          data-test-subj="confirmVisChangeModal"
+        >
+          <p>
+            <FormattedMessage
+              id="wizard.rightNav.changeVisType.modalDescription"
+              defaultMessage="Currently, changing the visualization type clears the existing selection of fields. Are you sure you want to change the visualization type?"
```
Comment (joshuarrrr) [inline src/plugins/wizard/public/application/components/right_nav.tsx:82]: This is a bit wordy for my taste - I think we could condense quite a bit.
```suggestion
              defaultMessage="Changing the visualization type will reset all field selections. Are you sure you want to continue?"
```

File: src/plugins/wizard/public/visualizations/common/expression_helpers.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+import { VisualizationState } from '../../application/utils/state_management';
+import { getAggService, getIndexPatterns } from '../../plugin_services';
+
+export const getAggExpressionFunctions = async (visualization: VisualizationState) => {
+  const { activeVisualization, indexPattern: indexId = '' } = visualization;
+  const { aggConfigParams } = activeVisualization || {};
+
+  const indexPatternsService = getIndexPatterns();
+  const indexPattern = await indexPatternsService.get(indexId);
+  const aggConfigs = getAggService().createAggConfigs(indexPattern, cloneDeep(aggConfigParams));
```
Comment (joshuarrrr) [inline src/plugins/wizard/public/visualizations/common/expression_helpers.ts:19]: An explanatory comment about why this is necessary might be helpful. It appears that we don't necessarily trust our `aggConfigParams` to be built with the correct `indexPattern`, but it's not clear from the code why that would be the case.

File: src/plugins/wizard/public/visualizations/common/expression_helpers.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    {}
+  );
+
+  // soon this becomes: const opensearchaggs = vis.data.aggs!.toExpressionAst();
+  const opensearchaggs = buildExpressionFunction<OpenSearchaggsExpressionFunctionDefinition>(
+    'opensearchaggs',
+    {
+      index: indexId,
+      metricsAtAllLevels: false,
+      partialRows: false,
```
Comment (joshuarrrr) [inline src/plugins/wizard/public/visualizations/common/expression_helpers.ts:32]: We should probably have a placeholder issue somewhere for supporting partial row aggs.

File: src/plugins/wizard/public/visualizations/vislib/area/area_vis_type.ts [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+          {
+            group: AggGroupNames.Buckets,
+            name: 'segment',
+            title: i18n.translate('visTypeVislib.area.segmentTitle', {
+              defaultMessage: 'X-axis',
+            }),
+            min: 0,
+            max: 1,
+            aggFilter: ['!geohash_grid', '!geotile_grid', '!filter', '!filters'],
+            defaults: { aggTypes: ['terms'] },
```
Comment (joshuarrrr) [inline src/plugins/wizard/public/visualizations/vislib/area/area_vis_type.ts:49]: This is more of a product decision, but I don't think this is the best default agg for an area chart.

File: src/plugins/wizard/public/visualizations/vislib/common/get_value_axes.ts [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+    scale: {
+      type: 'linear',
+      mode: 'normal',
+    },
+    show: true,
+    style: {},
+    title: {
+      text: y.label,
+    },
+    type: 'value',
```
Comment (joshuarrrr) [inline src/plugins/wizard/public/visualizations/vislib/common/get_value_axes.ts:30]: We also need a placeholder issue for exposing axes control to users

File: src/plugins/wizard/public/visualizations/vislib/histogram/histogram_vis_type.ts [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+        schemas: new Schemas([
+          {
+            group: AggGroupNames.Metrics,
+            name: 'metric',
+            title: i18n.translate('visTypeVislib.histogram.metricTitle', {
+              defaultMessage: 'Y-axis',
+            }),
+            min: 1,
+            max: 3,
+            aggFilter: ['!geo_centroid', '!geo_bounds'], // '!geo_centroid', '!geo_bounds'
```
Comment (joshuarrrr) [inline src/plugins/wizard/public/visualizations/vislib/histogram/histogram_vis_type.ts:37]: nit - can remove comment

File: src/plugins/wizard/public/visualizations/vislib/line/line_vis_type.ts [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+          {
+            group: AggGroupNames.Buckets,
+            name: 'segment',
+            title: i18n.translate('visTypeVislib.line.segmentTitle', {
+              defaultMessage: 'X-axis',
+            }),
+            min: 0,
+            max: 1,
+            aggFilter: ['!geohash_grid', '!geotile_grid', '!filter', '!filters'],
+            defaults: { aggTypes: ['terms'] },
```
Comment (joshuarrrr) [inline src/plugins/wizard/public/visualizations/vislib/line/line_vis_type.ts:49]: same as area, I think a better default agg would be better here.

**Review by joshuarrrr — APPROVED** [review]
Nice job on getting all the basics working! Some questions and callouts for future work, but I think this will give people a lot more freedom to play with the tool!

Now that we have all the functionality working, we'll want to also set aside some more time to get a lot of the defaults set, which will minimize user steps for common tasks.

File: test/functional/apps/wizard/_base.ts [hunk 10]
```diff
@@ -27,22 +28,33 @@ export default function ({ getService, getPageObjects }: FtrProviderContext) {
     });
 
     it('should show visualization when a field is added', async () => {
-      await PageObjects.wizard.addField('metric', 'Average', 'machine.ram');
-      const avgMachineRam = ['13,104,036,080.615', 'Average machine.ram'];
+      const expectedData = [2858, 2904, 2814, 1322, 2784];
+      await PageObjects.wizard.addField('metric', 'Count');
+      await PageObjects.wizard.addField('segment', 'Terms', 'machine.os.raw');
```
[the remark this answers is not in the document]
Comment (ashwin-pc) [reply test/functional/apps/wizard/_base.ts:33]: The previous test was for a single metric. This creates a bar chart since that's the default vis type now

File: src/plugins/wizard/public/application/components/right_nav.tsx (hunk 2 above)
Comment (ashwin-pc) [reply src/plugins/wizard/public/application/components/right_nav.tsx:76]: +1. I'll create a follow-up issue to consolidate them

File: src/plugins/wizard/public/application/_variables.scss (hunk 1 above)
Comment (ashwin-pc) [reply src/plugins/wizard/public/application/_variables.scss:10]: Oh yes, I'll remove that. Nice catch.

File: src/plugins/wizard/public/visualizations/common/expression_helpers.ts (hunk 4 above)
Comment (ashwin-pc) [reply src/plugins/wizard/public/visualizations/common/expression_helpers.ts:19]: I can add that. `AggConfigParams` does not actually have an index pattern. It's just a serialized array of aggregations and field types. We need to reconstruct `aggConfigs` which is an object using the agg service using the index pattern and serialized agg config params

File: src/plugins/wizard/public/visualizations/common/expression_helpers.ts (hunk 5 above)
Comment (ashwin-pc) [reply src/plugins/wizard/public/visualizations/common/expression_helpers.ts:32]: Given that we are migrating away from vislib, I kept the implementation intentionally light so that we have less doc migration issues once we have our final feature set. Im hoping that with this basic version we can get the feedback to identify what those features are, including partial buckets. I'll open a single issue to track all of that :)

File: src/plugins/wizard/public/visualizations/vislib/area/area_vis_type.ts (hunk 6 above)
Comment (ashwin-pc) [reply src/plugins/wizard/public/visualizations/vislib/area/area_vis_type.ts:49]: +1. Same as the above

File: src/plugins/wizard/public/visualizations/vislib/common/get_value_axes.ts (hunk 7 above)
Comment (ashwin-pc) [reply src/plugins/wizard/public/visualizations/vislib/common/get_value_axes.ts:30]: +1 Same as the above

File: src/plugins/wizard/public/visualizations/vislib/line/line_vis_type.ts (hunk 9 above)
Comment (ashwin-pc) [reply src/plugins/wizard/public/visualizations/vislib/line/line_vis_type.ts:49]: +1 same as the above

Comment (ashwin-pc) [reply src/plugins/wizard/public/visualizations/vislib/line/line_vis_type.ts:49]: Also, this is for the future, I want the default agg type to depend on the field type and not be a simple array like this. I will open an issue for that

File: test/functional/apps/wizard/_base.ts (hunk 10 above)
[the remark this answers is not in the document]
Comment (joshuarrrr) [reply test/functional/apps/wizard/_base.ts:33]: Cool, that's what I figured!

**Review by joshuarrrr — APPROVED** [review]

File: src/plugins/wizard/public/visualizations/metric/to_expression.ts [hunk 11]
```diff
@@ -3,18 +3,13 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import { cloneDeep } from 'lodash';
```
Comment (AMoo-Miki) [inline src/plugins/wizard/public/visualizations/metric/to_expression.ts:6]: Thanks you!

**Review by AMoo-Miki — APPROVED** [review]

**Review by bandinib-amzn — APPROVED** [review]

**Review by kavilla — APPROVED** [review]
works for me!
